eng/tools/TypeSpecValidation/src/TypeSpecValidation.ts

```diff
@@ -0,0 +1,48 @@
+import {simpleGit} from 'simple-git';
+import { exec } from "child_process";
+
+async function runCmd(cmd:string) {
+    console.log(`run command:${cmd}`)
+    const { err, stdout, stderr } = await new Promise((res) =>
+        exec(
+        cmd,
+        { encoding: "utf8", maxBuffer: 1024 * 1024 * 64 },
+        (err: unknown, stdout: unknown, stderr: unknown) =>
+            res({ err: err, stdout: stdout, stderr: stderr })
+        )
+    ) as any;
+    let resultString = stderr + stdout;
+    console.log("Stdout output:")
+    console.log(stdout)
+    if (stderr) {
+        console.log("Stderr output:")
+        console.log(stderr)
+    }
+    if (stderr || err) {
+        throw new Error(err);
+    }
+    return resultString as string;
+}
+    
+async function main() {
+    const args = process.argv.slice(2);
+    const folder = args[0];
+    console.log("Running TypeSpecValidation on folder:", folder);
+    const output = await runCmd(
+        `npx tsp compile ${folder}`
```

Comment:
Given we are just shelling out commands do we think using a script like PS or bash would be better then wrapping this in TS?

Reply:
For package managing this tool for other repos to use & for the possibility of adding this to the tsp compile script, I think TS might be better for these cases.